src/compiler/factory/nodeFactory.ts

```diff
@@ -4481,6 +4482,25 @@ export function createNodeFactory(flags: NodeFactoryFlags, baseFactory: BaseNode
         return node;
     }
 
+    // @api
+    function updateModuleDeclarationAndKeyword(
+        node: ModuleDeclaration,
+        modifiers: readonly ModifierLike[] | undefined,
+        name: ModuleName,
+        body: ModuleBody | undefined
+    ) {
+        if (node.flags & (NodeFlags.Ambient | NodeFlags.Namespace)) {
+            return updateModuleDeclaration(node, modifiers, name, body);
+        }
+
+        return node.modifiers !== modifiers
+            || node.name !== name
+            || node.body !== body
+            ? update(createModuleDeclaration(modifiers, name, body, node.flags | NodeFlags.Namespace), node)

```

Comment:
As discussed with Titian, we should be able to replace this final return with simply:
`return update(createModuleDeclaration(modifiers, name, body, node.flags | NodeFlags.Namespace), node)`